src/components/ClinicalFeatureTable.js

```diff
@@ -0,0 +1,287 @@
+import { Alert, Button, Collapse, Input, Label, Table } from 'reactstrap';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import Backend from '../services/backend';
+import { useKeycloak } from '@react-keycloak/web';
+import { CLINCAL_FEATURE_TYPES, CLINICAL_FEATURE_ENCODING } from '../config/constants';
+import { validateLabelOrClinicalFeaturesFile } from './utils/feature-utils.js';
+
+import './ClinicalFeatureTable.css';
+
+
+export default function ClinicalFeatureTable({
+  clinicalFeaturesColumns,
+  dataPoints,
+  isSavingClinicalFeatures,
+}) {
+  let { keycloak } = useKeycloak();
+
+  let [editableClinicalFeatures, setEditableClinicalFeatures] = useState({});
+
+  let [isManualClinFeaturesOpen, setisManualClinFeaturesOpen] = useState(true);
+  let [isAutoClinFeaturesOpen, setisAutoClinFeaturesOpen] = useState(false);
+
+  let [isClinicalFeatureFileValid, setisClinicalFeatureFileValid] = useState(null);
+  let [clinicalFeatureFileMessage, setclinicalFeatureFileMessage] = useState(null);
+  let fileInput = useRef(null);
+
+  const [editableClinicalFeatureDefinitions, setEditableClinicalFeatureDefinitions] = useState({
+    // "Age": { "Type": CLINCAL_FEATURE_TYPES[0], "Encoding": "None" },
+    // "Gender": { "Type": CLINCAL_FEATURE_TYPES[0], "Encoding": "Categorical" }
+  });
+
+  const handleInputChange = (e, feature_name, feature_type) => {
+    let editableClinicalFeatureDefinitionsToUpdate = { ...editableClinicalFeatureDefinitions };
+    editableClinicalFeatureDefinitionsToUpdate[feature_name][feature_type] = e.target.value;
+    setEditableClinicalFeatureDefinitions(editableClinicalFeatureDefinitionsToUpdate)
+  };
+
+  useEffect(() => {
```

Comment:
By convention, it is usual to put all the "hooks" (useState, useEffect, etc.) at the top of the file, before declaring other functions, maybe you can hoist your useEffect hooks above the "handleInputChange" function?